pandas/tests/arrays/test_array.py

```diff
@@ -393,6 +393,21 @@ def test_array_not_registered(registry_without_decimal):
     tm.assert_equal(result, expected)
 
 
+def test_datetime64_eq_string():
```

Comment:
this doesn't go here. move the test to pandas/tests/frame/methods/test_astype.py